esmvaltool/_task.py

```diff
@@ -134,18 +134,28 @@ def _format(value):
         return txt
 
     def _format_dict(name, dictionary):
-        """Format dict as NCL"""
-        lines = ['{} = True'.format(name)]
+        """Format dict as NCL (list of) logical(s) with attributes."""
+        if name in ('diag_script_info', 'config_user_info'):
+            syntax = ['{} = True', '{}@{} = {}']
+        else:
+            syntax = ['{}[i] = True', '{}[i]@{} = {}']
+        lines = [syntax[0].format(name)]
         for key, value in sorted(dictionary.items()):
-            lines.append('{}@{} = {}'.format(name, key, _format(value)))
+            lines.append(str[1].format(name, key, _format(value)))
         txt = '\n'.join(lines)
         return txt
 
     def _header(name):
-        """Delete any existing NCL variable known as `name`."""
-        return ('if (isvar("{name}")) then\n'
-                '    delete({name})\n'
-                'end if\n'.format(name=name))
+        """Create NCL header either as a logical or as a list of logicals."""
+        if name in ('diag_script_info', 'config_user_info'):
```

Comment:
Also here, yhe purpose of this function is convert *generic* python/yaml to NCL. It should not depend on the variable names.